components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/impl/WebAppAuthenticatorImpl.java

```diff
@@ -118,6 +118,8 @@ private boolean validateScopes(Message message, AccessTokenInfo tokenInfo) {
         String verb = (String) message.get(Message.HTTP_REQUEST_METHOD);
         String resource = path.substring(basePath.length() - 1);
         String[] scopes = tokenInfo.getScopes();
+        // Set the token info into the message context
+        message.getExchange().put(RestApiConstants.MESSAGE_EXCHANGE_TOKEN_INFO, tokenInfo);
```

Comment:
The existing constant USER_REST_API_SCOPES is related to only the scopes. Here I am adding all the token Info into(which include scopes) the message exchange, so that it could be used later.

Reply:
Why do we need all the token information? The problem is if you are using a new constant, you need to put it to basic authentication as well. Otherwise, it will break when someone invokes an API using basic authentication headers.

Reply:
Fixed with 0e39028